### PR TITLE
✨ add animateKeyFrames support

### PR DESCRIPTION
**Pull request by ahmed-yamany**
https://github.com/marcosgriselli/ViewAnimator/assets/58072774/3775cf8f-452c-4431-913b-4191ea0c5078

    The `animateKeyFrames` method animates a series of keyframes, each defined by an `Animation` object.

     Example usage:

   ```swift
  let angle: CGFloat = .pi / 8
     let animatins: [Animation] = [
         AnimationType.rotate(angle: -angle),
         AnimationType.rotate(angle: angle),
         AnimationType.rotate(angle: -angle),
         AnimationType.rotate(angle: angle),
         AnimationType.rotate(angle: 0)
     ]

     view.animateKeyFrames(animations: animations, initialAlpha: 0.5, finalAlpha: 1.0, delay: 0.2, duration: 2.0) {
         // Completion handler
         print("Animation completed!")
     }
```

### CODE REVIEW

**Comment by marcosgriselli** [comment]
Did not expect new features for this library. Thanks 🎉

File: Sources/ViewAnimator.swift [hunk 1]
```diff
@@ -97,7 +97,67 @@ public extension UIView {
...
+            for index in 0..<numberOfFrames {
+                let animation = animations[index]
+                let frameDurationStartTime = index == 0 ? 0.0 : Double(singleFrameDuration) * Double(index)
+                //
+                UIView.addKeyframe(withRelativeStartTime: frameDurationStartTime, relativeDuration: singleFrameDuration) {
+                    self?.transform = animation.initialTransform
+                    self?.alpha = initialAlpha == 1.0 ? 1 : frameDurationStartTime
+                }
+            }
+            //
```
Comment (marcosgriselli) [inline Sources/ViewAnimator.swift:153]: Same

**Review by marcosgriselli — APPROVED** [review]

File: Sources/ViewAnimator.swift (hunk 1 above)
Comment (ahmed-yamany) [reply Sources/ViewAnimator.swift:153]: if the initialAlpha equals 1  it must not animate the alpha

File: Sources/ViewAnimator.swift [hunk 2]
```diff
@@ -97,7 +97,67 @@ public extension UIView {
...
+                          completion: (() -> Void)? = nil) {
+        let numberOfFrames: Int = animations.count
+        let singleFrameDuration: Double = Double(1/Double(numberOfFrames))
+        //
+        alpha = initialAlpha
+        UIView.animateKeyframes(withDuration: duration + singleFrameDuration, delay: delay, options: options) { [weak self] in
+            for index in 0..<numberOfFrames {
+                let animation = animations[index]
+                let frameDurationStartTime = index == 0 ? 0.0 : Double(singleFrameDuration) * Double(index)
+                //
```
[the remark this answers is not in the document]
Comment (ahmed-yamany) [reply Sources/ViewAnimator.swift:147]: This line calculates the relative start time for the current keyframe. If it's the first keyframe (index 0), the relative start time is set to 0.0. For subsequent keyframes, the relative start time is calculated based on the singleFrameDuration and the index of the keyframe. This ensures that each keyframe starts at the appropriate time in the overall animation sequence

[the remark this answers is not in the document]
Comment (ahmed-yamany) [reply Sources/ViewAnimator.swift:147]: So, frameDurationStartTime represents the relative start time for each keyframe, allowing them to be staggered in time based on their index in the animations array.

[the remark this answers is not in the document]
Comment (marcosgriselli) [reply Sources/ViewAnimator.swift:150]: we should add the comment or remove the empty `//`

[the remark this answers is not in the document]
Comment (ahmed-yamany) [reply Sources/ViewAnimator.swift]: done

**Comment by ahmed-yamany** [comment]
https://github.com/marcosgriselli/ViewAnimator/assets/58072774/13e90df3-4c47-4e0e-b48f-cb4bad7d6d28

here are some animations made with this new feature to a tabBar

```swift
  func homeAnimate(_ view: UIView) {
        view.animateKeyFrames(animations: [
            AnimationType.vector(.init(dx: 10, dy: 0)),
            AnimationType.vector(.init(dx: 0, dy: 0)),
            AnimationType.vector(.init(dx: -10, dy: 0)),
            AnimationType.vector(.init(dx: 0, dy: 0)),
            AnimationType.zoom(scale: 0.5),
            AnimationType.zoom(scale: 1.3),
            AnimationType.zoom(scale: 1)
        ], initialAlpha: 1, duration: 1)
    }
```

```swift
   func locationAnimate(_ view: UIView) {
        view.animateKeyFrames(animations: [
            AnimationType.from(direction: .bottom, offset: -20),
            AnimationType.from(direction: .bottom, offset: 10),
            AnimationType.zoom(scale: 0.9),
            AnimationType.from(direction: .bottom, offset: -10),
            AnimationType.zoom(scale: 1),
            AnimationType.from(direction: .bottom, offset: 0)
        ], initialAlpha: 1, duration: 1)
    }
```

```swift
  func appointmentAnimate(_ view: UIView) {
        let angle: CGFloat = .pi / 8
        view.animateKeyFrames(animations: [
            AnimationType.zoom(scale: 1.5),
            AnimationType.rotate(angle: -angle),
            AnimationType.zoom(scale: 0.7),
            AnimationType.rotate(angle: 0),
            AnimationType.zoom(scale: 1.3),
            AnimationType.zoom(scale: 1)
        ], initialAlpha: 1, duration: 1)
    }
```

```swift
   func profileAnimate(_ view: UIView) {
        view.animateKeyFrames(animations: [
            AnimationType.zoom(scale: 0.9),
            AnimationType.zoom(scale: 0.7),
            AnimationType.zoom(scale: 0.5),
            AnimationType.rotate(angle: 90),
            AnimationType.rotate(angle: 180),
            AnimationType.rotate(angle: 270),
            AnimationType.rotate(angle: 360),
            AnimationType.rotate(angle: 0),
            AnimationType.zoom(scale: 1)
        ], initialAlpha: 1, duration: 1)
    }
```

**Pull request merged**